mobile/src/main/java/rs/readahead/washington/mobile/views/activity/VideoViewerActivity.kt

```diff
@@ -0,0 +1,540 @@
+    }
+
+    override fun onExportError(error: Throwable) {
+        showToast(R.string.gallery_toast_fail_exporting_to_device)
+    }
+
+    override fun onExportStarted() {
+        progressDialog = DialogsUtil.showProgressDialog(
+            this, getString(R.string.gallery_save_to_device_dialog_progress_expl)
+        )
+    }
+
+    override fun onExportEnded() {
+        hideProgressDialog()
+    }
+
+    override fun onMediaFileDeleted() {
+        MyApplication.bus().post(MediaFileDeletedEvent())
+        finish()
+    }
+
+    override fun onMediaFileDeletionError(throwable: Throwable) {
+        showToast(R.string.gallery_toast_fail_deleting_files)
+    }
+
+    override fun onMediaFileRename(vaultFile: VaultFile) {
+        toolbar!!.title = vaultFile.name
+        this.vaultFile = vaultFile
+        MyApplication.bus().post(VaultFileRenameEvent())
+    }
+
+    override fun onMediaFileRenameError(throwable: Throwable) {
+        //TODO CHECK ERROR MSG WHEN RENAME
+        DialogUtils.showBottomMessage(
+            this, getString(R.string.gallery_toast_fail_deleting_files), true
+        )
+    }
+
+    override fun getContext(): Context {
+        return this
+    }
+
+    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
+        // Show the controls on any key event.
+        simpleExoPlayerView.showController()
+        // If the event was not handled then see if the player view can handle it as a media key event.
+        return super.dispatchKeyEvent(event) || simpleExoPlayerView.dispatchMediaKeyEvent(event)
+    }
+
+    private fun shareMediaFile() {
+        if (vaultFile == null) {
+            return
+        }
+        if (vaultFile?.metadata != null) {
+            showShareWithMetadataDialog()
+        } else {
+            startShareActivity(false)
+        }
+    }
+
+    private fun startShareActivity(includeMetadata: Boolean) {
+        if (vaultFile == null) {
+            return
+        }
+        MediaFileHandler.startShareActivity(this, vaultFile, includeMetadata)
+    }
+
+    private fun initializePlayer() {
+        val needNewPlayer = player == null
+        if (needNewPlayer) {
+            player = ExoPlayer.Builder(this).build()
+            player?.apply {
+                binding.playerView.player
+                playWhenReady = shouldAutoPlay
+            }
+
+            simpleExoPlayerView.setPlayer(player)
+        }
+        if (needNewPlayer || needRetrySource) {
+            if (intent.hasExtra(VIEW_VIDEO) && intent.extras != null) {
+                val vaultFile = intent.extras!![VIEW_VIDEO] as VaultFile?
+                if (vaultFile != null) {
+                    this.vaultFile = vaultFile
+                    toolbar?.title = vaultFile.name
+                    setupMetadataMenuItem(vaultFile.metadata != null)
+                }
+            }
+            val mediaFileDataSourceFactory = MediaFileDataSourceFactory(
+                this, vaultFile!!, null
```

Comment:
`MediaFileDataSourceFactory` requires a non null instance of `VaulFile` and here it's not the case.